indico/modules/events/papers/templates/_contributions.html

```diff
@@ -1,10 +1,11 @@
 {% macro render_paper_section(contribution, show_paper=True) %}
+    {% set can_access = (contribution.abstract and contribution.abstract.submitter == session.user) or contribution.is_user_associated(session.user) %}
```

Comment:
I think you could also use `paper.can_submit`. Check it out in the `Paper` model.